### PR TITLE
Refactor to Nextra instead of Gatsby

### PR DESCRIPTION
**Pull request by dimaMachina**

### CODE REVIEW

**Comment by Urigo** [comment]
Hi @graphql/tsc and dear community

We need your help in reviewing 2 things here:

1. Migration of the current website from Gatsby to Nextra
2. The new GraphQL Conf 2024 website

Some important clarifications:
1. We want to get as much feedback as possible. Share as much as you can. Regressions from the current graphql.org while also ideas on how to make it better
2. We'll merge this before we'll solve all the issues you will raise. We will merge when the new website will be **not worse** then the current one. The main goal here is to simply change the infrastructure (even though @dimaMachina came up with some new wins as during his work) 
3. We'll take all of your feedback and create issues for the points that we can improve after merging this change and build a roadmap out of it.
4. There is a separate discussion around getting a designer to create a new version of the website, that is a separate effort from what you see here.

Looking forward for your feedback

**Comment by Urigo** [comment]
Hi @graphql/tsc and dear community

We need your help in reviewing 2 things here:

1. Migration of the current website from Gatsby to Nextra
2. The new GraphQL Conf 2024 website

Please review using [this link](https://graphql-github-io-git-fork-dimam-b98301-the-graph-ql-foundation.vercel.app?_vercel_share=h6h6G13gbBHT799xWc6aBxZ6UX8TB030)

Some important clarifications:
1. We want to get as much feedback as possible. Share as much as you can. Regressions from the current graphql.org while also ideas on how to make it better
2. We'll merge this before we'll solve all the issues you will raise. We will merge when the new website will be **not worse** then the current one. The main goal here is to simply change the infrastructure (even though @dimaMachina came up with some new wins as during his work) 
4. We'll take all of your feedback and create issues for the points that we can improve after merging this change and build a roadmap out of it.
5. There is a separate discussion around getting a designer to create a new version of the website, that is a separate effort from what you see here.

Thank you @danstarns for the inspiration for the events and meetups page!

Looking forward for your feedback

File: website/theme.config.tsx
Comment (fotoetienne) [inline website/theme.config.tsx]: This link is broken. Should it be `/conf/2024` ?

**Review by fotoetienne — APPROVED** [review]
Looks great!

**Comment by Keweiqu** [comment]
Thanks @Urigo! I checked the review link on my phone and noticed there is some minor resizing issue on the GraphQL.org home page. It doesn’t resize correctly for the mobile screen when there is a code block. 

The graphql conf page looks great! Haven’t spotted any issue

**Comment by tazsingh** [comment]
Looks awesome! Great work @dimaMachina @Urigo & all 🙌 

Per points 2 & 3 above, should I create an issue or comment here?

I note the Community section is missing:
- London GraphQL - https://guild.host/london-graphql
- GraphQL New York - https://guild.host/graphql-new-york
- GraphQL Bali - https://guild.host/graphql-bali
- GraphQL Taipei - https://guild.host/graphql-taipei

We can also power this list of communities as well as the events themselves via a direct feed, so it's always up to date, in addition to the JSON. But I take that as a larger refactoring. I believe Step 1 could be to add this to the current JSON format for the time being.

Thanks and let me know how I can help with this!

**Comment by erikwrede** [comment]
I'm liking the fresh look!
These are my rather opinionated insights and suggestions. They're not always ordered based on urgency, so feel free to prioritize/reconsider points as deemed necessary.

<details>
<summary> Header Bar </summary>
 
Header Bar
The header bar appears rather busy now, this might be somewhat confusing for visitors. Here are my suggestions:
- Consider highlighting the GraphQL Conf with a button (for example, pink) as a Call-to-Action. Eliminate the dropdown and redirect to the 2024 page - provide a conspicuous button for locating the 2023 conference page on the 2024 page. This simplifies the process and directly guides visitors to the most recent and relevant conference.
- Future thought: Code + Learn + FAQ could be consolidated into a Resources & Getting Started Tab. For first-time visitors, the content of "Code" is not instantly clear.
- The "external" arrow on Spec is not necessary in my opinion.
- Consider increasing the font size of the header to text-md for enhanced readability, considering the font weight and contrast ratio - this adjustment is only plausible if we reduce the number of entries.
</details>

<details>
<summary> Landing Page </summary>
  
I'm really liking the design revamp on this page! There's just one minor point: the black carousel-like banner overlaps with the header on wide screens, creating a drop-shadow effect when scrolled to the top. Interestingly, this doesn't occur when the screen width is reduced. Refer to the header bar screenshot for details.
</details>

<details>
<summary> Code </summary>

Lacking the explanatory text from the older version, the purpose of this page may not be instantly discernible to visitors. I have a few usability suggestions:
- Reintroduce some text/header indicating that this is the go-to place for discovering the next GraphQL framework, library, or tooling.
- Consider categorizing the filters by programming language and type of tool.
- Make the framework names clickable again (redirecting to GitHub/Homepage), though retain the GitHub button, it has a nice appeal.
- Text search seems to have an issue:
  
Text Search
  
In the future, it would also be worthwhile to consider embedding the framework logos. This could help break the monotony of the list style. 
</details>

<details>
<summary> Community Resources </summary>
  
Community Resources
The cards are consuming a significant amount of vertical space. We could think about compacting them a bit like this for familiar logos (not a perfect example, but hopefully, the idea is clear 🙂):

Familiar Logos

Alternatively, introducing a breakpoint to transition into a four-column layout on desktop (depending on resizing) might be a good idea:
image
  
Currently, the remainder of the page somewhat resembles a SEO link and resource list. Some future curation efforts could help transform this into an excellent jump-off point and assist with discoverability.
</details>

<details>
<summary> Events and Meetups </summary>
  
It's great to finally have this on the main organization page. This is well-done for a first version. Going forward, introducing a location-based search for "Meetups & Events near me" can significantly improve user experience.
</details>

Overall, I didn't encounter any significant issues that are "worse than the current version". Thanks for putting all the effort into this!

File: website/theme.config.tsx
Comment (dimaMachina) [reply website/theme.config.tsx]: this link is broken only in prod, because Next.js' rewrites/redirects doesn't work for static exported website https://github.com/dimaMachina/graphql.github.io/blob/1057765e20da6afd9cdc313218dea77c1304c985/website/next.config.js#L43-L50

before releasing new website we'll make all redirects/rewrites declared in next.js config working!

**Comment by dimaMachina** [comment]
@tazsingh Thank you for your feedback 😍!

> We can also power this list of communities as well as the events themselves via a direct feed, so it's always up to date, in addition to the JSON. But I take that as a larger refactoring. I believe Step 1 could be to add this to the current JSON format for the time being.

We'll improve this in the future thanks to Next.js SSG, for now I just hardcoded your suggested meetups

https://github.com/dimaMachina/graphql.github.io/blob/1057765e20da6afd9cdc313218dea77c1304c985/website/src/components/meetups.ts#L2-L41

you can take a look at the last preview:

https://graphql-github-rliggs852-the-graph-ql-foundation.vercel.app/community/events/

**Comment by benjie** [comment]
The copyright year is 2023 for some reason; other than that - nothing further to add! Thanks for the work on this!

**Comment by bignimbus** [comment]
@Keweiqu mentioned mobile resizing issues - here's a screencap here of what I saw on my phone (iPhone 12 mini, Firefox):

https://github.com/graphql/graphql.github.io/assets/3976692/a2aa6ae6-ae5e-4a61-81de-fc75ae2407ca

Thank you for your work on this! 🩷

File: website/src/components/meetups.ts
Comment (danstarns) [inline website/src/components/meetups.ts]:
```suggestion
export const meetups = [
  {
    node: {
      id: "london",
      name: "London GraphQL",
      link: "https://guild.host/london-graphql/events",
      city: "London",
      latitude: 51.51,
      longitude: -0.12,
    },
  },
  {
    node: {
      id: "new-york",
      name: "GraphQL New York",
      link: "https://guild.host/graphql-new-york/events",
      city: "New York",
      latitude: 40.73,
      longitude: -73.94,
    },
  },
  {
    node: {
      id: "bali",
      name: "GraphQL Bali",
      link: "https://guild.host/graphql-bali/events",
      city: "Bali",
      latitude: -8.41,
      longitude: 115.19,
    },
  },
  {
    node: {
      id: "bangkok",
      name: "GraphQL Bangkok",
      link: "https://www.meetup.com/graphql-bangkok",
      city: "Bangkok",
      latitude: 13.75,
      longitude: 100.50,
    },
  },
  {
    node: {
      id: "bangalore",
      name: "GraphQL Bangalore",
      link: "https://www.meetup.com/graphql-bangalore",
      city: "Bangkok",
      latitude: 12.97,
      longitude: 77.59,
    },
  },
  {
    node: {
      id: "sydney",
      name: "GraphQL Sydney",
      link: "https://graphql.sydney/",
      city: "Sydney",
      latitude: -33.86,
      longitude: 151.20,
    },
  },
  {
    node: {
      id: "taipei",
      name: "GraphQL Taipei",
      link: "https://guild.host/graphql-taipei/events",
      city: "Taipei",
      latitude: 25.11,
      longitude: 121.6,
    },
  },
  {
    node: {
      id: "20446155",
      name: "GraphQL TLV",
      latitude: 32.07,
      longitude: 34.77,
      city: "Tel Aviv-Yafo",
      state: "",
      country: "il",
      link: "https://www.meetup.com/graphql-tlv",
    },
  },
  {
    node: {
      id: "31704787",
      name: "GraphQL Zurich",
      latitude: 47.38,
      longitude: 8.54,
      city: "Zürich",
      state: "",
      country: "ch",
      link: "https://www.meetup.com/graphql-zurich",
    },
  },
  {
    node: {
      id: "29671383",
      name: "GraphQL Seattle",
      latitude: 47.61,
      longitude: -122.33,
      city: "Seattle",
      state: "WA",
      country: "us",
      link: "https://www.meetup.com/graphql-seattle",
    },
  },
  {
    node: {
      id: "31255122",
      name: "GraphQL SF",
      latitude: 37.78,
      longitude: -122.42,
      city: "San Francisco",
      state: "CA",
      country: "us",
      link: "https://www.meetup.com/sf-graphql",
    },
  },
  {
    node: {
      id: "37102548",
      name: "GraphQL Gathering EU in Berlin",
      latitude: 52.52,
      longitude: 13.38,
      city: "Berlin",
      state: "",
      country: "de",
      link: "https://www.meetup.com/graphql-gathering-eu",
    },
  },
  {
    node: {
      id: "20208222",
      name: "Amsterdam GraphQL Meetup",
      latitude: 52.37,
      longitude: 4.89,
      city: "Amsterdam",
      state: "",
      country: "nl",
      link: "https://www.meetup.com/amsterdam-graphql-meetup",
    },
  },
  {
    node: {
      id: "32869172",
      name: "Copenhagen GraphQL Meetup Group",
      latitude: 55.68,
      longitude: 12.57,
      city: "Copenhagen",
      state: "",
      country: "dk",
      link: "https://www.meetup.com/copenhagen-graphql-meetup-group",
    },
  },
  {
    node: {
      id: "27845373",
      name: "GraphQL Hamburg",
      latitude: 53.55,
      longitude: 10,
      city: "Hamburg",
      state: "",
      country: "de",
      link: "https://www.meetup.com/graphql-hamburg",
    },
  },
  {
    node: {
      id: "20451795",
      name: "ATX GraphQL",
      latitude: 30.27,
      longitude: -97.71,
      city: "Austin",
      state: "TX",
      country: "us",
      link: "https://www.meetup.com/atx-graphql",
    },
  },
]
```

Comment (danstarns) [reply website/src/components/meetups.ts]: Added Bangalore, Bangkok and Sydney. 

FYI most of this data already lives here: https://graphql-events.org/api/meetups

Happy to help you use our API and shape the data, else, if you would like this as the source of truth we can donate/forward graphql-events.org to this page.

File: CONTRIBUTING.md
Comment (danstarns) [inline CONTRIBUTING.md:49]: Please confirm if its `yarn start` at the root or `cd website && pnpm start`.

File: website/src/components/index-page/index.tsx
Comment (danstarns) [inline website/src/components/index-page/index.tsx:10]: Screenshot 2024-03-19 at 10 35 42

Something jarring about each slide here, it looks like there is more space/padding on the right to the left - shifting the view to the left.

File: website/src/components/index-page/powerful-tools.tsx
Comment (danstarns) [inline website/src/components/index-page/powerful-tools.tsx:1]: The video here is cut off at the bottom and breaks the flow of rounded edges. 

Screenshot 2024-03-19 at 10 44 53

Comment (danstarns) [reply website/src/components/index-page/powerful-tools.tsx:1]: Also the second video, loops and displays a empty slide. 


https://github.com/graphql/graphql.github.io/assets/35999252/11492423-e74b-426e-a06e-f0c4657db2bd

File: website/src/components/index-page/hero.tsx
Comment (danstarns) [inline website/src/components/index-page/hero.tsx]: The animations here are rather slow, and given they are first text to render this could effect page ranking score and user experience due to first meaningfull paint. Maybe we just remove the animations at the top of the page. 

https://web.dev/articles/vitals

Screenshot 2024-03-19 at 10 58 26

**Review by danstarns — COMMENTED** [review]
Looking good, some comments from me 😎

File: website/src/components/index-page/powerful-tools.tsx
Comment (dimaMachina) [reply website/src/components/index-page/powerful-tools.tsx:1]: > The video here is cut off at the bottom and breaks the flow of rounded edges.
> 
> Screenshot 2024-03-19 at 10 44 53

this is tricky to record because when you hover some link you have this popup 
image
any ideas how to hide it?

**Comment by dimaMachina** [comment]
> @Keweiqu mentioned mobile resizing issues - here's a screencap here of what I saw on my phone (iPhone 12 mini, Firefox):
> 
>  trim.58215B96-EB5C-49D2-BC7D-46BC621E8411.MOV 
> Thank you for your work on this! 🩷

thank you! mobile issues on landing page/code page were fixed

**Comment by dimaMachina** [comment]
> The copyright year is 2023 for some reason; other than that - nothing further to add! Thanks for the work on this!

fixed copyright in footer and faq page!

**Comment by Urigo** [comment]
We are live!
Thanks everyone for the valuable feedback!

It doesn't mean we finished solving and working on all your suggestions, this is just the start of many improvements to come
Keep suggesting improvements, creating issues and also sending PRs to make our community's website the best it could be!

**Comment by Urigo** [comment]
and mostly thanks to @dimaMachina for this amazing migration project!

**Comment by Urigo** [comment]
and thanks @YassinEldeeb for taking care of Vercel deployments, redirects and making sure we've covered the important things ❤️

**Pull request merged**
